contracts/TopShot.cdc

```diff
@@ -654,10 +665,95 @@ pub contract TopShot: NonFungibleToken {
+                        createEmptyCollectionFunction: (fun (): @NonFungibleToken.Collection {
+                            return <-TopShot.createEmptyCollection()
+                        })
+                    )
+                case Type<MetadataViews.NFTCollectionDisplay>():
+                    let media = MetadataViews.Media(
+                        file: MetadataViews.HTTPFile(
+                            url: "https://nbatopshot.com/static/img/top-shot-logo-horizontal-white.svg"
+                        ),
+                        mediaType: "image/svg+xml"
+                    )
+                    return MetadataViews.NFTCollectionDisplay(
+                        name: "NBA TopShot",
+                        description: "NBA Top Shot is your chance to own, sell, and trade official digital collectibles of the NBA and WNBA's greatest plays and players",
+                        externalURL: MetadataViews.ExternalURL("https://nbatopshot.com"),
+                        squareImage: media,
+                        bannerImage: media,
+                        socials: {}
+                    )
+                case Type<MetadataViews.Traits>():
+                    let traitDictionary: {String: AnyStruct} = {
+                        "Date Of Moment": TopShot.getPlayMetaDataByField(playID: self.data.playID, field: "DateOfMoment"),
+                        "Play Category": TopShot.getPlayMetaDataByField(playID: self.data.playID, field: "PlayCategory"),
+                        "Play Type": TopShot.getPlayMetaDataByField(playID: self.data.playID, field: "PlayType"),
+                        "Series Number": TopShot.getSetSeries(setID: self.data.setID),
+                        "Set Name": TopShot.getSetName(setID: self.data.setID),
+                        "Serial Number": self.data.serialNumber
+                    }
+                    return MetadataViews.dictToTraits(dict: traitDictionary, excludedNames: [])
             }
 
             return nil
         }   
+
+        // Functions used for computing MetadataViews 
+
+        // getMomentURL 
+        // Returns: The computed external url of the moment
+        pub fun getMomentURL(): String {
+            return "https://nbatopshot.com/moment/".concat(self.id.toString())
+        }
+        // getEditionName Moment's edition name a combination of the Moment's setName and playID
+        // `setName: #playID`
+        pub fun getEditionName() : String {
+            let setName: String = TopShot.getSetName(setID: self.data.setID) ?? ""
+            let editionName = setName.concat(": #").concat(self.data.playID.toString())
+            return editionName
+        }
+
+        // imageURL will build the url for the image associated with the moment based on its metadata
+        //
+        // Returns: The computed external url of a medium sized image associated with the moment
+        pub fun imageURL(): String {
```

Comment:
Is there a license assumption here that people should be aware of? My guess is that just because an image is here doesn't mean that everyone is allowed to show it. Might be worth adding that into the comments if applicable

Reply:
This is a good call out - I had been asking around if there were any partnership related concerns with adding image urls on chain and not found immediate concerns. Going to double check with product/partnerships to see if any caveats would be appropriate here.